docs/apache-airflow/extra-packages-ref.rst

```diff
@@ -18,17 +18,24 @@
 Reference for package extras
 ''''''''''''''''''''''''''''
 
-Here's the list of all the extra dependencies of Apache Airflow.
+Airflow has a number of optional "extras" that you can use to add features to your installation when you
+are installing Airflow. Those extras are a good way for the users to manage their installation, but also
+they are useful for contributors to airflow when they want to contribute some of the featuers - including
+optional integrations of Airflow - via providers.
 
-The entries with ``*`` in the ``Preinstalled`` column indicate that those extras (providers) are always
-pre-installed when Airflow is installed.
+,, warning::
+
+    Traditionally in Airflow some of the extras used `.` and `_` to separate the parts of the extra name.
+    This was not PEP-685 normalized name and we opted to change it to to `-` for all our extras, Expecting that
+    PEP-685 will be implemented in full by `pip` and other tools. Currently the normalization is happening
+    anyway, but `pip` shows warning when `_` or `-` are used, due to old packaging version used (January 2023).
+    The work is in progress to change it in `this issue <https://github.com/pypa/pip/issues/11445>` so this
+    is anticipated that it will be fixed soon.
+
+    TODO(potiuk): decide whether to do it. In the current proposal we changed everything to `_`.
```

Comment:
Mhm, would be good to be "future ready" - do you know if this is backwards-compatible? How about all users trying to install via `pip install apache-airflow[apache.spark]` - if we change to `_` or `-` will this be corrected and resolved by PIP or do we need to publish an alias as well to keep backwards-compatability?

Reply:
Yes. This is what I've decided now (see previous comment). Let's see what @uranusjr thinks about it. I believe it is "pretty much fully backwards compatible" to use `-` in all airflow extras,

Reply:
Resolving it as the status is kept in the previous comment.